### DISCUSSION TITLE
GPIO Inputs Availiblity

### DISCUSSION THREAD

**Discussion by MrPaulAR** [forkineye/ESPixelStick#517] [Q&A] [answered]
Are GPIO inputs available on either of the ESP platforms?

What I'm hoping to do is use GPIO inputs to trigger (or cycle) different FSEQ to be played with no network access.

**Comment by MartinMueller2003** [comment, marked as answer]
There are some pins available on the larger ESP32 platforms. Not many on the ESP8266 platforms. Yes we could easily add triggering fseq files based on a pulse from the GPIOs.

**Reply by MrPaulAR** [reply]
Is it worthwhile making this a feature request for esp32 to have configurable GPIO inputs as triggers?

**Comment by MartinMueller2003** [comment]
Yes I would create a feature request for it. It is an interesting feature and most of the code to do it is already there.

Think about how the buttons would be used. What happens at the end of the song? Does it stop? Play again? What happens if a song is playing and the same button is pressed again? Does it ignore the press? Does it restart the song? What if a song is playing and a different button is pressed? Do you move to the next song or remember the request and play the selected song next? How deep would the queue of songs be?

These are the types of decisions that should be in your request.

**Reply by MrPaulAR** [reply]
Done.. https://github.com/forkineye/ESPixelStick/issues/518

**Comment by MartinMueller2003** [comment]
You did not answer any of the questions above. :(
